doc/ensemble.rst

```diff
@@ -0,0 +1,739 @@
+argument with the path to the text file, `configure_params.txt`:
+
+.. code-block:: python
+
+    # Attach the file to the Model instance
+    model.attach_generator_files(to_configure="path/to/configure_params.txt")
+
+When we launch the Ensemble using ``Experiment.start()``, SmartSim reads the values within the text
+file, registered the value of `perm_strategy` which we set to `all_perm`, and will create three
+different Models within the Ensemble.
+
+=====================
+ML Models and Scripts
+=====================
+--------
+Overview
+--------
+SmartSim supports sending TorchScript functions, scripts, and
+TF, TF-lite, PT, or ONNX models to the database at runtime
+prior to the execution of ensemble members for use within the workload.
+The ``Ensemble API`` provides a subset of helper functions that support
+these capabilities:
+
+* ``Ensemble.add_ml_model()`` : Load a TF, TF-lite, PT, or ONNX model into the DB at runtime.
+* ``Ensemble.add_function()`` : Launch a TorchScript function with each ensemble member.
+* ``Ensemble.add_script()`` : Launch a TorchScript with each ensemble member.
+
+---------
+AI Models
+---------
+.. _load_ml_model:
+When configuring an ensemble, users can instruct SmartSim to load
+TensorFlow (TF), TensorFlow Lite (TF-lite), PyTorch (PT), or ONNX
+models dynamically to the database (colocated or standard). Machine Learning (ML) models added
+are loaded prior to the execution of ensemble members and therefore
+ready for use when a ensemble member is invoked. SmartSim users may
+providing the model in memory or specifying its file path via the
+``Ensemble.add_ml_model()`` Ensemble API helper function.
+
+When specifying an ML model using ``Ensemble.add_ml_model()``, the
+following arguments are offered:
+
+-  `name` (str): Key used to store the model within the ensemble.
+-  `model` (str | bytes | None): Model name in memory.
+-  `model_path` (str): File path to the serialized model.
+-  `backend` (str): Name of the model backend (TORCH, TF, TF-LITE, ONNX).
+-  `device` (str, optional): Name of the device for execution (defaults to "CPU").
+-  `batch_size` (int, optional): Batch size for execution (defaults to 0).
+-  `min_batch_size` (int, optional): Minimum batch size for model execution (defaults to 0).
+-  `tag` (str, optional): Additional tag for model information (defaults to an empty string).
+-  `inputs` (list[str], optional): Names of model inputs (TF only, defaults to None).
+-  `outputs` (list[str], optional): Names of model outputs (TF only, defaults to None).
+
+These arguments provide details to add and configure
+ML models within the ensemble simulation.
+
+Example: Loading an In-Memory ML Model to the Ensemble
+------------------------------------------------------
+In this example, we demonstrate how to instruct SmartSim to load
+an in-memory ML model into the database at ensemble runtime. It's
+important to note that in-memory ML models are supported for
```

Comment:
- "it's important to note..." could be a note/warning block

- is it meant to say "in-memory ML models are supported for non-colo..." OR "in-memory ML models are supported _ONLY_ for non-colo..."

Reply:
* Could we just say "standard" instead of "non-colo"? It feels more readable to avoid the negation and just say what we mean. 
 
The closing blurb says standard:

`When the ensemble is started via ``Experiment.start()``, the ML model will be loaded to the standard orchestrator that is launched prior to the start of the ensemble.`